server/py/framework/db/sqldb/db.py

```diff
@@ -3901,16 +3901,38 @@ def _calculate_mm_functions_counters(
         )  # filter duplications
         labels = label_set(labels)
         query = self._add_labels_filter(session, query, Function, labels)
+        query = query.with_entities(
+            Function.project.label("project"), Function.state.label("function_state")
+        )
+        subq = query.subquery()
+        query = session.query(
+            subq.c.project,
+            func.count(
+                case(
+                    (
+                        subq.c.function_state
+                        == mlrun.common.schemas.FunctionState.ready,
+                        1,
+                    )
+                )
+            ).label("running_count"),
+            func.count(
+                case(
+                    (
+                        subq.c.function_state
+                        == mlrun.common.schemas.FunctionState.error,
+                        1,
+                    )
+                )
+            ).label("failed_count"),
+        ).group_by(subq.c.project)
+        results = query.all()
```

Comment:
Query looks good!
For these kinds of complicated queries I'd probably add a unit test to make sure it behaves how you expect it to. Fill the inmemory/file sqlite db with some data and then run the method to see you get the expected numbers